packages/account/src/account.ts

```diff
@@ -782,6 +811,27 @@ export class Account extends AbstractAccount implements WithAddress {
       : undefined;
   }
 
+  /** @hidden * */
+  private async assembleTx(
+    transactionRequest: ScriptTransactionRequest,
+    quantities: CoinQuantity[] = []
+  ): Promise<{ transactionRequest: ScriptTransactionRequest; gasPrice: BN }> {
+    const outputQuantities = transactionRequest.outputs
+      .filter((o) => o.type === OutputType.Coin)
+      .map(({ amount, assetId }) => ({ assetId: String(assetId), amount: bn(amount) }));
+
+    transactionRequest.gasLimit = bn(0);
+    transactionRequest.maxFee = bn(0);
+
+    const { assembledRequest, gasPrice } = await this.provider.assembleTx({
+      request: transactionRequest,
+      accountCoinQuantities: mergeQuantities(outputQuantities, quantities),
+      feePayerAccount: this,
+    });
+
+    return { transactionRequest: assembledRequest as ScriptTransactionRequest, gasPrice };
```

Comment:
```suggestion
    return { assembledRequest: assembledRequest as ScriptTransactionRequest, gasPrice };
```

I think all `assembleTx` methods should return the same signature.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3747#discussion_r2013995198